spec/models/user_spec.rb

```diff
@@ -119,26 +119,26 @@
         },
       )
     end
+  end
 
-    context "when changing role" do
-      described_class.roles.reject { |role| role == "trial" }.each do |_role_name, role_value|
-        it "updates org when changing from trial to #{role_value}" do
-          user = create(:user, :with_trial)
+  context "when changing role" do
+    described_class.roles.reject { |role| role == "trial" }.each do |_role_name, role_value|
+      it "updates org when changing from trial to #{role_value}" do
+        user = create(:user, role: :trial)
 
-          expect(Form).to receive(:update_org_for_creator).with(user.id, user.organisation.slug)
+        expect(Form).to receive(:update_org_for_creator).with(user.id, user.organisation.slug)
 
-          user.role = role_value
-          user.save!
-        end
+        user.role = role_value
+        user.update_org
```

Comment:
It's kind of weird that we have to call `update_org` after changing the role... I get that we're updating the org for the forms, but when we have these two lines next to each other it's not obvious why they go together. I think it would be quite easy to assume that they're not related, when in fact they are.

I wonder if we calling the new method `update_user_forms` instead, would make things clearer...